sycl/doc/extensions/ExplicitSIMD/dpcpp-explicit-simd.md

```diff
@@ -0,0 +1,776 @@
+Examples:
+##### 1D surface block store.
+T - element type, n - vector length, acc - SYCL global buffer accessor,
+offset - byte offset in the buffer, vals - vector value to store
+```cpp
+template <typename T, int n, typename AccessorTy>
+void block_store(AccessorTy acc, uint32_t offset, simd<T, n> vals);
+```
+##### 2D media block load.
+T - element type, m and n - block dimensions, acc - SYCL image2D accessor,
+x and y - image coordinates
+```cpp
+template <typename T, int m, int n, typename AccessorTy, unsigned plane = 0>
+simd<T, m * n> media_block_load(AccessorTy acc, unsigned x, unsigned y);
+```
+
+#### Local address space allocation and access
+Examples:
+
+##### SLM scatter.
+T - element type, n - vector length (16 or 32), vals - vector value to store,
+offsets - byte offsets in the local memory, pred - store mask
+```cpp
+template <typename T, int n>
+typename std::enable_if<(n == 16 || n == 32), void>::type
+slm_store(simd<T, n> vals, simd<uint32_t, n> offsets, simd<uint16_t, n> pred = 1);
+```
+<br>
+
+### Private Global Variables.
+
+Explicit SIMD extension supports "private global" variables - file scope
+variables in private address space (similar to thread-local variables on host).
+These variables have 1 copy per work-item (which maps to a single SIMD thread in
+ESP) and are visible to all functions in the translation unit. Conceptually they
+map to SPIRV variable with private storage class. Private globals can be bound
+to a specific byte offset within the GRF. To mark a file scope variable as
+private global, the `INTEL_GPU_PRIVATE` attribute is used,
+`INTEL_GPU_REGISTER` is used to bind it the register file:
+
+```cpp
+INTEL_GPU_PRIVATE INTEL_GPU_REGISTER(32) simd<int, 16> vc;
+```
+<br>
+
+## Examples
+### Vector addition (USM)
+```cpp
+#include <iostream>
+#include <CL/sycl.hpp>
+#include <sycl_esimd.hpp>
+
+using namespace cl::sycl;
+
+int main(void) {
+  constexpr unsigned Size = 128;
+  constexpr unsigned VL = 32;
+  constexpr unsigned GroupSize = 2;
+
+  queue q;
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+  auto ctxt = q.get_context();
+  float* A = static_cast<float*>(malloc_shared(Size*sizeof(float), dev, ctxt));
+  float* B = static_cast<float*>(malloc_shared(Size*sizeof(float), dev, ctxt));
+  float* C = static_cast<float*>(malloc_shared(Size*sizeof(float), dev, ctxt));
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = B[i] = i;
+  }
+
+  // iteration space
+  nd_range<1> Range(range<1>(Size/VL), range<1>(GroupSize));
+
+  auto e = q.submit([&](handler &cgh) {
+    cgh.parallel_for<class Test>(
+      Range, [=](nd_item<1> i) [[intel::sycl_explicit_simd]] {
+
+      auto offset = i.get_global_id(0) * VL;
+      sycl::intel::gpu<float, VL> va = sycl::intel::gpu::flat_block_load<float, VL>(A + offset);
+      sycl::intel::gpu<float, VL> vb = sycl::intel::gpu::flat_block_load<float, VL>(B + offset);
+      sycl::intel::gpu<float, VL> vc = va + vb;
```

Comment:
3 `auto` in LHS? I guess you want to sell your extension... :-)

Reply:
changed to explicit type